tls/s2n_cipher_suites.c

```diff
@@ -837,17 +837,18 @@ const struct s2n_cipher_preferences cipher_preferences_test_all_tls12 = {
  * in order of IANA value. Exposed for the "test_all_fips" cipher preference list.
  */
 static struct s2n_cipher_suite *s2n_all_fips_cipher_suites[] = {
-    &s2n_rsa_with_3des_ede_cbc_sha,           /* 0x00,0x0A */
-    &s2n_rsa_with_aes_128_cbc_sha,            /* 0x00,0x2F */
-    &s2n_rsa_with_aes_256_cbc_sha,            /* 0x00,0x35 */
-    &s2n_rsa_with_aes_128_cbc_sha256,         /* 0x00,0x3C */
-    &s2n_rsa_with_aes_256_cbc_sha256,         /* 0x00,0x3D */
+    &s2n_dhe_rsa_with_aes_128_cbc_sha,        /* 0x00,0x33 */
+    &s2n_dhe_rsa_with_aes_256_cbc_sha,        /* 0x00,0x39 */
     &s2n_dhe_rsa_with_aes_128_cbc_sha256,     /* 0x00,0x67 */
     &s2n_dhe_rsa_with_aes_256_cbc_sha256,     /* 0x00,0x6B */
-    &s2n_rsa_with_aes_128_gcm_sha256,         /* 0x00,0x9C */
-    &s2n_rsa_with_aes_256_gcm_sha384,         /* 0x00,0x9D */
     &s2n_dhe_rsa_with_aes_128_gcm_sha256,     /* 0x00,0x9E */
     &s2n_dhe_rsa_with_aes_256_gcm_sha384,     /* 0x00,0x9F */
+    &s2n_tls13_aes_128_gcm_sha256,            /* 0x13,0x01 */
+    &s2n_tls13_aes_256_gcm_sha384,            /* 0x13,0x02 */
+    &s2n_ecdhe_ecdsa_with_aes_128_cbc_sha,    /* 0xC0,0x09 */
+    &s2n_ecdhe_ecdsa_with_aes_256_cbc_sha,    /* 0xC0,0x0A */
+    &s2n_ecdhe_rsa_with_aes_128_cbc_sha,      /* 0xC0,0x13 */
+    &s2n_ecdhe_rsa_with_aes_256_cbc_sha,      /* 0xC0,0x14 */
     &s2n_ecdhe_ecdsa_with_aes_128_cbc_sha256, /* 0xC0,0x23 */
```

Comment:
CCM ciphersuites are also approved. But they are marginally used. I suggest you do not include them.

Reply:
s2n-tls doesn't support CCM.